pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -575,19 +601,10 @@ def fn(item: ResultMetric) -> None:
 
         apply_to_collection(self, ResultMetric, fn)
 
-    def extract_batch_size(self, batch: Any) -> int:
-        try:
-            batch_size = extract_batch_size(batch)
-        except RecursionError:
-            batch_size = 1
-        self.batch_size = batch_size  # the setter converts it to `Tensor`
-        return batch_size
-
     def to(self, *args: Any, **kwargs: Any) -> "ResultCollection":
         """Move all data to the given device."""
         self.update(apply_to_collection(dict(self), (torch.Tensor, Metric), move_data_to_device, *args, **kwargs))
 
-        self._batch_size = self._batch_size.to(*args, **kwargs)
```

Comment:
This would still be required with current_batch_size if we were to move the ResultCollection on CPU mid batch iteration.

Reply:
I wonder do we even need to move the batch_size to the device at all. It's a scalar and wherever it's used, it's used along with other tensors which are already on their respective devices.

```bash
>>> a = torch.tensor(10.).cuda()
>>> a + 7
torch(17., device='cuda:0')
>>> a * 7
torch(70., device='cuda:0')
```

Reply:
This is done to ensure there is no memory left on the GPU. We have tests for this and they would fail.